BXMSDOC-5549: Add management console sections. (#2350)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/bpmn/proc-management-console-using.adoc

{URL_CONFIGURING_KOGITO}#con-data-index-service_kogito-configuring[_{CONFIGURING_KOGITO}_].
endif::[]
ifdef::KOGITO-COMM[]
xref:con-data-index-service_kogito-configuring[].
endif::[]
* The `pom.xml` file of your {PRODUCT} project contains the following dependency for the process management add-on. This add-on enables the Management Console to interact with the process data through the add-on REST endpoint `/management/processes`. For more information about the process management add-on, see xref:con-bpmn-process-management-addon_kogito-developing-process-services[].
+
.Project dependency to enable process management REST operations
[source,xml]
----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>process-management-addon</artifactId>
</dependency>
----
* The `application.properties` file of your {PRODUCT} project contains the following system property for the location where the {PRODUCT} service is deployed, such as `\http://localhost:8080`. This property enables the Management Console to generate the URLs to execute the REST operations from the process management add-on.
+
.Application property for {PRODUCT} service location
[source,subs="+quotes"]
----

> kogito.service.url=http://__HOST__:__PORT__

Reviewer: @sterobin I missed another requirement here, which is a temporary one for developer preview and will be removed in the next versions. CORS needs to be enabled at this stage to allow the remote REST calls. See an example here for a Quarkus based project: https://github.com/kiegroup/kogito-examples/blob/0.9.x/kogito-travel-agency/travels/src/main/resources/application.properties#L1

Author: Done. Updated as follows per gchat convo.

```
The `application.properties` file of your {PRODUCT} project contains the following system properties for the location where the {PRODUCT} service is deployed, such as `\http://localhost:8080`, and for Quarkus Cross-Origin Resource Sharing (CORS) support. These properties enables the Management Console to generate the URLs to execute the REST operations from the process management add-on.
+
.Application properties for REST URLs
[source,subs="+quotes"]
----
kogito.service.url=http://__HOST__:__PORT__
quarkus.http.cors=true
----
```

----

.Procedure
. Go to the https://repository.jboss.org/org/kie/kogito/management-console/[`management-console`] artifacts page, select the latest release of the {PRODUCT} Management Console, and download the `management-console-__VERSION__-runner.jar` file to a local directory.
. In a command terminal, navigate to the directory location of the downloaded `management-console-__VERSION__-runner.jar` file and enter the following command to run the Management Console:
+
--
.Running the Management Console
[source,subs="+quotes"]
----
$ java -Dquarkus.http.port=8280 -jar management-console-__VERSION__-runner.jar
----

> NOTE: The Management Console uses the default Data Index Service port 8280. If you modified this port in your Data Index Service instance, you must also modify the port in the Management Console properties.

Reviewer: ```suggestion
NOTE: The Management Console uses the default Data Index Service port 8180. If you modified this port in your Data Index Service instance, you must also modify the port in the Management Console properties.
```

Would be good to mention the property here, example: 
`-Dkogito.dataindex.http.url=http://localhost:8180`

Author: Fixed to this per gchat convo.

```
[NOTE]
====
The default port for the Management Console is 8080, but this example specifies port 8280 to avoid conflicts with the example {PRODUCT} service running at port 8080.

Also, the Management Console uses the default Data Index Service port 8180. If you modified this port in your Data Index Service instance, you must also modify the port in the Management Console properties by using the start-up property `-Dkogito.dataindex.http.url=http://__HOST__:__PORT__` when you run the Management Console.
====
```

To change the logging level of the Management Console, such as for debugging, you can specify the following start-up properties:

.Modifying Management Console logging level for debugging
[source,subs="+quotes"]
----
$ java  \
  -Dquarkus.log.console.level=DEBUG -Dquarkus.log.category.\"org.kie.kogito\".min-level=DEBUG  \
  -Dquarkus.log.category.\"org.kie.kogito\".level=DEBUG  \
  -Dquarkus.http.port=8280  \
  -jar management-console-__VERSION__-runner.jar
----

In an OpenShift environment, you can use the {PRODUCT} command-line interface (CLI) or the OpenShift web console to add the Management Console to your {PRODUCT} infrastructure:

.Adding the Management Console to your OpenShift infrastructure using the {PRODUCT} CLI
[source]
----
kogito install management-console
----

.{PRODUCT} Management Console instance on OpenShift web console
image::kogito/openshift/kogito-management-console-instance.png[Image of Kogito Management Console instance on OpenShift]
--

> . In a web browser, navigate to `http://localhost:8280/ProcessInstances` to open the Management Console. If you modified the default Data Index Service URL, use the modified location.

Reviewer: ```suggestion
. In a web browser, navigate to `http://localhost:8280/` to open the Management Console. If you modified the default Data Index Service URL, use the modified location.
```

Author: Fixed

+
NOTE: The only bulk operation currently supported is *Abort*.

+
.Expanded process with subprocesses
image::kogito/bpmn/kogito-management-console-subprocesses.png[Image of expanded process with subprocesses in Management Console]
+
.Details for a selected process name
image::kogito/bpmn/kogito-management-console-process-details.png[Image of process details in Management Console]

* *Domain Explorer*: Use this page to view and filter data that is generated from your process instances in the available {PRODUCT} services, or _domains_, such as the `Travels` and `VisaApplications` domains in this example.
+
.Domain explorer with available domains
image::kogito/bpmn/kogito-management-console-domain-explorer.png[Image of domain explorer in Management Console]
+
.VisaApplications domain data
image::kogito/bpmn/kogito-management-console-domain-explorer-visas.png[Image of VisaApplications domain data in Management Console]
+

> .Filters for VisaApplications domain data

Reviewer: These are the avaiable attributes of VisaApplications domain, users can select which ones to display.

Author: Done, changed to "Attributes for ..." and said "sort" instead of "filter".

image::kogito/bpmn/kogito-management-console-domain-explorer-visas-filters.png[Image of filters for VisaApplications domain data in Management Console]
+

> .Filtered data for VisaApplications domain

Reviewer: Same comment as above, this is more like a result list, showing the selected attributes from a specific doman. Just concerned to call it filters/filtered because the actual filtering feature is coming next.

Author: Done, changed to "Sorted data for ..."

quarkus.http.cors=true
----

.Procedure
. Go to the https://repository.jboss.org/org/kie/kogito/management-console/[`management-console`] artifacts page, select the latest release of the {PRODUCT} Management Console, and download the `management-console-__VERSION__-runner.jar` file to a local directory.
. In a command terminal, navigate to the directory location of the downloaded `management-console-__VERSION__-runner.jar` file and enter the following command to run the Management Console:
+
--
.Running the Management Console
[source,subs="+quotes"]
----
$ java -Dquarkus.http.port=8280 -jar management-console-__VERSION__-runner.jar
----

[NOTE]
====
The default port for the Management Console is 8080, but this example specifies port 8280 to avoid conflicts with the example {PRODUCT} service running at port 8080.

Also, the Management Console uses the default Data Index Service port 8180. If you modified this port in your Data Index Service instance, you must also modify the port in the Management Console properties by using the start-up property `-Dkogito.dataindex.http.url=http://__HOST__:__PORT__` when you run the Management Console.
====

> To change the logging level of the Management Console, such as for debugging, you can specify the following start-up properties:

Reviewer 2: Just a consideration. "such as" Stands out more as an example. 

May be better as: To change the logging level of the Management Console (for example, for debugging) you can specify the following start-up properties: 

I am really not sure if it makes a difference. I often use "such as" and feel as if I should not. Feel free to correct me.

Author: I see. I use such as as a smoother and less interruptive alternative to "for example". I use the later when I want to really emphasize something as an example. So personally I'm okay with the usage, but am happy to hear more of a counter argument.

.Modifying Management Console logging level for debugging
[source,subs="+quotes"]
----
$ java  \
  -Dquarkus.log.console.level=DEBUG -Dquarkus.log.category.\"org.kie.kogito\".min-level=DEBUG  \
  -Dquarkus.log.category.\"org.kie.kogito\".level=DEBUG  \
  -Dquarkus.http.port=8280  \
  -jar management-console-__VERSION__-runner.jar
----

> In an OpenShift environment, you can use the {PRODUCT} command-line interface (CLI) or the OpenShift web console to add the Management Console to your {PRODUCT} infrastructure:

Reviewer 2: Do you need "you can"?

Author: I know what you mean. A lot of times it can be ditched, but other times I like it because it conveys that it's an option or at least that it's not a command or step. Kind of softens the instruction to be more informative than imperative.

.Adding the Management Console to your OpenShift infrastructure using the {PRODUCT} CLI
[source]
----
kogito install management-console
----

.{PRODUCT} Management Console instance on OpenShift web console
image::kogito/openshift/kogito-management-console-instance.png[Image of Kogito Management Console instance on OpenShift]
--

> . In a web browser, navigate to `http://localhost:8280` to open the Management Console. If you modified the default Data Index Service URL, use the modified location.

Reviewer: ```suggestion
. In a web browser, navigate to `http://localhost:8280` to open the Management Console. If you modified the default Management Console port, use the modified location.
```

Author: Changed to this per gchat convo:

>In a web browser, navigate to http://localhost:8280 to open the Management Console. If you modified **the configured Management Console port**, use the modified location.

+
--
On OpenShift, navigate to the route URL for the Management Console instance.

.{PRODUCT} Management Console
image::kogito/bpmn/kogito-management-console.png[Image of Kogito Management Console]

In this example, the Management Console displays data for the `kogito-travel-agency` example application in the https://github.com/kiegroup/kogito-examples[`kogito-examples`] repository in GitHub.
--
. In the Management Console, use the following pages from the left menu to interact with your process instances and data:
+
--

> * *Process Instances*: Use this page to view and filter process and subprocess instances by status or business key. You can select a specific process instance name to view process details or abort the process, or select the check box for all relevant instances to perform a bulk abort operation.

Reviewer 2: Again, I try to remove "You can" where possible for minimalism. This may change the desired user approach if this is optional. Up to you. I won't add another note about "You can" :) 

Select a specific process instance name to view process details or..

Author: I agree and I'm guilty of using it when not necessary. Again here though, I think removing it, although better for minimalism, changes the messaging a bit. "Select a specific process ... or abort the process" seems very black and white and imperative. Also makes it seem like those are you only two options, but then immediately the next sentence gives you another option. So saying what "you can" do softens and opens the instruction, as opposed to hardening and closing it. That said, you mention a key point that I need to bear in mind and make sure I don't go nuts with it. Will be sure to use it judiciously. Thanks.
